nengo_loihi/neurons.py

```diff
@@ -231,7 +235,65 @@ def nengo_build_nif(model, nif, neurons):
 
 
 if nengo_dl is not None:  # noqa: C901
+    class TFSpikeNoiseGenerator(object):
```

Comment:
Is the reason that we can't use Nengo's ensemble noise parameter for this that the noise is dependent on the input current?  Assuming that I'm understanding that correctly, maybe we should consider reworking the ensemble noise builder so that we could support that.

Reply:
The ensemble noise is noise on the input current/voltage. This is the opposite kind of noise, noise on the output. It really only makes sense in the case of rate neurons, but I agree, it would be nice if Nengo supported it in some way (though I also don't want to clutter up the API, since I think this is only of interest to a small subset of users).

Reply:
Since this is just for rate neurons, and we know the nonlinearity, it seems like there should be some equivalent noise on the input side that would produce the same effect as noise on the output.  I haven't thought about that too carefully though 🤷‍♂️ 

In terms of API, maybe that'd look like a `noise` parameter on the `Connection`?  Definitely not a high priority at the moment though, agreed.

Reply:
So, for inputs greater than the firing threshold, the tuning curve is a one-to-one function. That means that it has a well defined inverse that you could apply to the output noise to get equivalent input noise. However, it's going to be messy, and you'll end up undoing all that computation when you pass things through the nonlinearity anyway, so why do it? As you might have noticed, the noise equations I'm using are quite specialized distributions (developed in my thesis), not a simple normal or uniform distribution.

You could put it on the connection, or you could allow Ensembles to have both input and output noise. I think the main use case is training in nengo_dl, though, so we might even want to consider some sort of config option that's nengo_dl specific.

Reply:
> However, it's going to be messy, and you'll end up undoing all that computation when you pass things through the nonlinearity anyway, so why do it?

Well the advantage would be that you could use the existing Ensemble noise parameter, rather than having to hack things into the neuron model.  But agreed that it's not worth worrying too much about it for this one case.